realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -481,6 +495,17 @@ public Builder rxFactory(RxObservableFactory factory) {
             return this;
         }
 
+        /**
+         * Sets the initial data in {@link io.realm.Realm}. This transaction will be executed only for the first time
+         * when database file is created or while migrating the data when {@link Builder#deleteRealmIfMigrationNeeded()} is set.
+         *
+         * @param transaction transaction to execute.
+         */
+        public Builder initialData(Realm.Transaction transaction) {
```

Comment:
My opinion is that supporting `Realm` only is fine. Most apps will use model classes and thereby `Realm`. Of course, it is important that we state it clearly in the documentation.